platform/darwin/src/MGLPolygon.mm

```diff
@@ -155,6 +167,22 @@ - (NSUInteger)hash {
     return hash;
 }
 
+- (CLLocationCoordinate2D)coordinate {
+    CLLocationCoordinate2D centroid;
+    
+    mbgl::Polygon<double> geometry;
+    for (MGLPolygon *polygon in self.polygons) {
+        geometry.push_back(polygon.ring);
```

Comment:
A multipolygon can represent multiple disjoint polygons, but this code gloms them together into a single polygon geometry. I believe that would be invalid according to the GeoJSON specification.

I belatedly realized that `polylabel()` only accepts an `mbgl::geometry::polygon`, not an `mbgl::geometry::multipolygon`. As I pointed out in https://github.com/mapbox/mapbox-gl-native/issues/7070#issuecomment-294022436, returning the centroid of an exotic shape like a multipolygon would be a low priority compared to some of the other functionality implemented here. For now, let’s just calculate the centroid of the first polygon in the shape and return that.